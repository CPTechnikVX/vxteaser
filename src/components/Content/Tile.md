```html
<tile link="https://www.visit-x.net/de/magazin/wiesn-gaudi/2018/">
  <panel>
    <row>
      <button link="https://www.visit-x.net/de/magazin/wiesn-gaudi/2018/">Zum Event</button>
    </row>
  </panel>
  <panel theme="light">
    <row>
      <headline type="main">
        <text>Wiesn </text>
        <text color="special">Gaudi - </text>
        <text color="highlight">sch√∂n</text>
      </headline>
    </row>
    <row>
      <headline>This is text</headline>
    </row>
  </panel>
</tile>
```
