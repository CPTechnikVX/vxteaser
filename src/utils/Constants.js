const classNames = {
	TeaserBanner:             'teaser-banner',
	TeaserBannerNoSkew:       'teaser-banner teaser--has-no-skew',
	TeaserBannerContent:      'teaser-banner__content',
	TeaserBannerContentInner: 'teaser-banner__content__inner',
	TeaserButton:             'teaser-button',
	TeaserSuite:              'teaser-suite',
	TeaserSuiteItem:          'teaser-suite__item',
	TeaserPoint:              'teaser-point',
	TeaserPointContainer:     'teaser-point-container',
};

export {
	classNames,
};