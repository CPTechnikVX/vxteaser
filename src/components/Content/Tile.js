import React       from 'react';
import PropTypes   from 'prop-types';
import classnames  from 'classnames';
import Constants   from '../../utils/Constants';
import TileContent from './TileContent';
import ClickEvent  from '../../utils/ClickEvent';

/**
 * Main Tile component
 */
export default class Tile extends React.PureComponent {
	constructor(props) {
		super(props);

		// bind
		this.onClickFn = this.onClickFn.bind(this);
	}

	onClickFn(event) {
		event.preventDefault();
		event.stopPropagation();

		this.props.onClickFn(new ClickEvent(event, this, {
			link: this.props.link,
		}));
	}

	render() {
		const {children, config, modifier, theme, windowWidth} = this.props;

		if (config) {
			const classList = [];
			classList.push(Constants.ClassName.Tile);

			const styleObj = {
				backgroundImage: `url('${config.fixedHeights[0]['backgroundUrl']}')`,
			};

			if (theme) {
				classList.push('-theme-' + theme);
			}

			if (modifier) {
				classList.push(modifier);
			}

			return (
				<div className={classnames(classList)} style={styleObj} onClick={this.onClickFn}>
					<TileContent config={config} windowWidth={windowWidth}>{children}</TileContent>
				</div>
			);
		} else {
			return null;
		}
	}
}

Tile.propTypes = {
	/** @ignore */
	config:      PropTypes.object,
	/** @ignore */
	children:    PropTypes.node,
	modifier:    PropTypes.string,
	/** @ignore */
	onClickFn:   PropTypes.func,
	link:        PropTypes.string,
	theme:       PropTypes.string,
	/** @ignore */
	windowWidth: PropTypes.number,
};
