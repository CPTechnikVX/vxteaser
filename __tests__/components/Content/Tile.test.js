import React      from 'react';
import {shallow}  from 'enzyme';
import Tile       from '../../../src/components/Content/Tile';
import Constants  from '../../../src/utils/Constants';
import classnames from 'classnames';

describe('render tile', () => {
    const windowWidth = 1300;
    const config      = {
        aspectRatio: [
            {
                backgroundUrl: 'url0',
            },
        ],
    };

    test('empty tile', () => {
        const wrapper = shallow(
                <Tile />
        );
        expect(wrapper).toMatchSnapshot();
    });
    test('tile content', () => {
        const wrapper = shallow(
                <Tile config={config}></Tile>
        );

        expect(wrapper.find('img').prop('src')).toBe(config.aspectRatio[0]['backgroundUrl']);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('TileContent')).toHaveLength(1);
        expect(wrapper.find('.' + classnames(Constants.ClassName.Tile).split(' ').join('.'))).toHaveLength(1);
    });
    test('tile theme', () => {
        const wrapper = shallow(
                <Tile config={config} theme={'white'}></Tile>
        );
        expect(wrapper).toMatchSnapshot();
    });
    test('tile pass props to children', () => {
        const wrapper = shallow(
                <Tile config={config} theme={'white'} windowWidth={windowWidth}></Tile>
        );
        expect(wrapper.find('TileContent').prop('windowWidth')).toBe(windowWidth);
    });
});
