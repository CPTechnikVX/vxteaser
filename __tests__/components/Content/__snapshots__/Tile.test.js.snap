// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render tile empty tile 1`] = `""`;

exports[`render tile tile content 1`] = `
<div
  className="vxteaser-tile"
  onClick={[Function]}
  style={
    Object {
      "backgroundImage": "url('url0')",
    }
  }
>
  <TileContent
    config={
      Object {
        "fixedHeights": Array [
          Object {
            "backgroundUrl": "url0",
          },
        ],
      }
    }
  />
</div>
`;

exports[`render tile tile theme 1`] = `
<div
  className="vxteaser-tile -theme-white"
  onClick={[Function]}
  style={
    Object {
      "backgroundImage": "url('url0')",
    }
  }
>
  <TileContent
    config={
      Object {
        "fixedHeights": Array [
          Object {
            "backgroundUrl": "url0",
          },
        ],
      }
    }
  />
</div>
`;
